freeipa/src/main/java/com/sequenceiq/freeipa/service/AltusMachineUserService.java

```diff
@@ -113,14 +142,34 @@ public DataBusCredential getOrCreateDataBusCredentialIfNeeded(Stack stack) throw
         } else {
             LOGGER.debug("Databus credential does not exist for the stack, it will be created ...");
         }
-        Optional<AltusCredential> altusCredential = createMachineUserWithAccessKeys(stack, telemetry);
+        Optional<AltusCredential> altusCredential = createDatabusMachineUserWithAccessKeys(stack, telemetry);
         return storeDataBusCredential(altusCredential, stack);
     }
 
+    public MonitoringCredential getOrCreateMonitoringCredentialIfNeeded(Stack stack) throws IOException {
```

Comment:
this should be return with optional I think as the caller start with a null check